### PR TITLE
Fix #16788: Optimize feed query performance issues introduced in 1.4.2

### LINKED ISSUE #16788: Activity feeds, mentions, KPI and home page responding very slow after upgrade to OpenMetaData 1.4.2
**Affected module**
It impacts home page response. very slow, always taking 10-15 secs

**Describe the bug**
Performance degrade while loading activity feeds, mentions & home page KPI 

**To Reproduce**
Upgrade to 1.4.2
Shared more information here: https://openmetadata.slack.com/archives/C02B6955S4S/p1718802959748509

**Expected behavior**
Ideally it should be instant as happened before

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8] 1.4.2
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by sonika-shah**
### Describe your changes:

Fixes #16788

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: bootstrap/sql/migrations/native/1.4.4/mysql/schemaChanges.sql [hunk 1]
```diff
@@ -1 +1,13 @@
-ALTER TABLE user_entity ADD UNIQUE (name);
\ No newline at end of file
+ALTER TABLE user_entity ADD UNIQUE (name);
+
+-- Adding a new column 'taskAssigneesIds' to improve query performance by avoiding the use of JSON_EXTRACT in queries
+-- This column is generated to store the 'id' values extracted from the 'taskAssignees' JSON array, which helps in optimizing search and filtering operations.
+ALTER TABLE thread_entity
+ADD COLUMN taskAssigneesIds VARCHAR(1000) GENERATED ALWAYS AS (JSON_EXTRACT(taskAssignees, '$[*].id'));
```
Comment (harshach) [inline bootstrap/sql/migrations/native/1.4.4/mysql/schemaChanges.sql:6]: varchar(1000) size is enough? what if there are 50 users?

Comment (sonika-shah) [reply bootstrap/sql/migrations/native/1.4.4/mysql/schemaChanges.sql:6]: no ..will not work for 50 max 20, will change to TEXT type

Comment (harshach) [reply bootstrap/sql/migrations/native/1.4.4/mysql/schemaChanges.sql:6]: not sure if TEXT is efficient in indexing

**Comment by sonika-shah** [comment]
Closing the pr . Fixed here -  https://github.com/open-metadata/OpenMetadata/pull/16862

**Pull request closed without merging**